diff --git a/src/image.c b/src/image.c
index 43220758324..d64034045a1 100644
--- a/src/image.c
+++ b/src/image.c
@@ -11751,7 +11751,7 @@ init_svg_functions (void)
 #if LIBRSVG_CHECK_VERSION (2, 48, 0)
   LOAD_DLL_FN (library, rsvg_handle_set_stylesheet);
 #endif
-#if LIBRSVG_CHECK_VERSION (2, 58, 0)
+#if LIBRSVG_CHECK_VERSION (2, 59, 0)
   LOAD_DLL_FN (library, rsvg_handle_get_pixbuf_and_error);
 #else
   LOAD_DLL_FN (library, rsvg_handle_get_pixbuf);
@@ -12354,7 +12354,7 @@ svg_load_image (struct frame *f, struct image *img, char *contents,
 
   /* We can now get a valid pixel buffer from the svg file, if all
      went ok.  */
-#if LIBRSVG_CHECK_VERSION (2, 58, 0)
+#if LIBRSVG_CHECK_VERSION (2, 59, 0)
   pixbuf = rsvg_handle_get_pixbuf_and_error (rsvg_handle, &err);
   if (err) goto rsvg_error;
 #else
